Core/Modules/DataAggregation.c

```diff
@@ -6,14 +6,15 @@
  */
 #include "DataAggregation.h"
 #include "ADCDriver.h"
+#include "HallDriver.h"
 
 // Global Variables
 SystemData_t SystemData;
 Events_t EventFlags;
 
 speed_t SystemGetSpeed()
 {
-	return SystemData.speed;
+	return HallGetSpeed();
```

Comment:
the purpose of the data aggregator is to act as a database. it should not be performing any functions or need to import any drivers. The task of reading sensors should be done in system task at a fixed rate and then values should be updated to data aggregator.